client/app/components/EditParameterSettingsDialog.jsx

```diff
@@ -0,0 +1,204 @@
+      onOk={onSaveClicked}
+      okText={isNew ? 'Add Parameter' : 'Save'}
+      okButtonProps={{
+        loading: saving,
+        disabled: isNew && (newNameFeedback.isError || !newName) || param.title === '',
+      }}
+    >
+      <Form layout="horizontal">
+        {isNew && (
+          <Form.Item
+            required
+            label="Keyword"
+            help={newNameFeedback.text}
+            validateStatus={newNameFeedback.isError ? 'error' : ''}
+            {...formItemProps}
+          >
+            <Input onChange={e => setNewName(e.target.value)} />
+          </Form.Item>
+        )}
+        <Form.Item label="Title" {...formItemProps}>
+          <Input
+            value={isNull(param.title) ? getDefaultTitle(newName) : param.title}
+            onChange={e => setParam({ ...param, title: e.target.value })}
+          />
+        </Form.Item>
+        <Form.Item label="Type" {...formItemProps}>
+          <Select value={param.type} onChange={type => setParam({ ...param, type })}>
+            <Option value="text">Text</Option>
+            <Option value="number">Number</Option>
+            <Option value="enum">Dropdown List</Option>
+            <Option value="query">Query Based Dropdown List</Option>
+            <Option disabled key="dv1">
+              <Divider className="select-option-divider" />
+            </Option>
+            <Option value="date">Date</Option>
+            <Option value="datetime-local">Date and Time</Option>
+            <Option value="datetime-with-seconds">Date and Time (with seconds)</Option>
+            <Option disabled key="dv2">
+              <Divider className="select-option-divider" />
+            </Option>
+            <Option value="date-range">Date Range</Option>
+            <Option value="datetime-range">Date and Time Range</Option>
+            <Option value="datetime-range-with-seconds">Date and Time Range (with seconds)</Option>
+          </Select>
+        </Form.Item>
+        {includes(['date', 'datetime-local', 'datetime-with-seconds'], param.type) && (
+          <Form.Item label=" " colon={false} {...formItemProps}>
+            <Checkbox
+              defaultChecked={param.useCurrentDateTime}
+              onChange={e => setParam({ ...param, useCurrentDateTime: e.target.checked })}
+            >
+              Default to Today/Now if no other value is set
+            </Checkbox>
+          </Form.Item>
+        )}
+        {param.type === 'enum' && (
+          <Form.Item label="Values" help="Dropdown list values (newline delimeted)" {...formItemProps}>
+            <Input.TextArea
+              rows={3}
+              value={param.enumOptions}
+              onChange={e => setParam({ ...param, enumOptions: e.target.value })}
+            />
+          </Form.Item>
+        )}
+        {param.type === 'query' && (
+          <Form.Item label="Query" help="Select query to load dropdown values from" {...formItemProps}>
+            <Select
```

Comment:
We have query search here, in "Add Widget Dialog" and in new alert page. Maybe time to have a <QuerySearch> component?

In the "Add Widget Dialog" we use a bit of a different UI for searching queries (input + list of results), but I don't see a reason not to switch to a Select element there as well.

Reply:
(btw, no need to introduce it to the other places now, just use it here for starters)